### PR TITLE
[9.3.0][xauth] Ansible class extended

### LINKED ISSUE #441: [QUESTION]: Are there any option to use one session ID for authentication
### How can the team help?
I would like to know if there are any  options to use a config file that has the authentication/sessionID instead of initiating an session on every ansible task for OME?
**Details: ?**
Currently we are foreseeing an issue when it comes to hitting the maximum API sessions if we have multiple tasks that will use the ansible modules for OME. We can probably increase the threshold but I want to put it out there if there are any possible solutions of using a single authentication code for one playbook  execution.

Thank you in advance and would appreciate any answers.

### PR DESCRIPTION
**Pull request by jagadeeshnv**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### OUTPUT

```paste below

```
##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

### CODE REVIEW

**Comment by ansible-collections-svc** [comment]
Can one of the admins verify this patch?

File: plugins/modules/idrac_certificates.py [hunk 1]
```diff
@@ -248,8 +248,8 @@
 import base64
 import os
 from datetime import datetime
-from ansible.module_utils.basic import AnsibleModule
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_certificates.py]: Remove extra line

File: plugins/modules/idrac_diagnostics.py [hunk 2]
```diff
@@ -344,8 +344,8 @@
 
 import json
 import os
-from ansible.module_utils.basic import AnsibleModule
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_diagnostics.py]: Remove extra line

File: plugins/modules/idrac_network_attributes.py [hunk 3]
```diff
@@ -320,11 +320,10 @@
 import time
 from urllib.error import HTTPError, URLError
```
Comment (rajshekarp87) [inline plugins/modules/idrac_network_attributes.py]: Remove extra line

File: plugins/modules/idrac_network_attributes.py [hunk 4]
```diff
@@ -320,11 +320,10 @@
...
 
-from ansible.module_utils.basic import AnsibleModule
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_network_attributes.py]: Remove extra line

File: plugins/modules/idrac_server_config_profile.py [hunk 5]
```diff
@@ -527,8 +527,8 @@
 import json
 from datetime import datetime
 from os.path import exists
-from ansible.module_utils.basic import AnsibleModule
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_server_config_profile.py]: Remove extra line

File: plugins/modules/idrac_storage_volume.py [hunk 6]
```diff
@@ -281,8 +281,8 @@
 import operator
 from urllib.error import HTTPError, URLError
 from copy import deepcopy
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
-from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, IdracAnsibleModule
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_storage_volume.py]: Remove extra line

File: plugins/modules/idrac_user_info.py [hunk 7]
```diff
@@ -116,8 +116,8 @@
 from ssl import SSLError
 from ansible.module_utils.six.moves.urllib.error import URLError, HTTPError
 from ansible.module_utils.urls import ConnectionError, SSLValidationError
-from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, idrac_auth_params
-from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.dellemc.openmanage.plugins.module_utils.idrac_redfish import iDRACRedfishAPI, IdracAnsibleModule
+
```
Comment (rajshekarp87) [inline plugins/modules/idrac_user_info.py]: Remove extra line

File: plugins/modules/idrac_redfish_storage_controller.py [hunk 8]
```diff
@@ -1001,8 +1000,8 @@ def main():
         "job_wait_timeout": {"required": False, "type": "int", "default": 120},
         "size": {"required": False, "type": "int"}
     }
-    specs.update(redfish_auth_params)
-    module = AnsibleModule(
+    # specs.update(redfish_auth_params)
```
Comment (rajshekarp87) [inline plugins/modules/idrac_redfish_storage_controller.py]:
```suggestion

```

File: tests/unit/plugins/modules/test_idrac_license.py [hunk 9]
```diff
@@ -736,8 +736,8 @@ def test_main(self, mocker):
 
         # Mock the necessary functions and objects
         mocker.patch(MODULE_PATH + 'get_argument_spec', return_value={})
-        mocker.patch(MODULE_PATH + 'idrac_auth_params', {})
-        mocker.patch(MODULE_PATH + 'AnsibleModule', return_value=module_mock)
+        # mocker.patch(MODULE_PATH + 'idrac_auth_params', {})
```
Comment (rajshekarp87) [inline tests/unit/plugins/modules/test_idrac_license.py]:
```suggestion

```

File: tests/unit/plugins/modules/test_idrac_diagnostics.py [hunk 10]
```diff
@@ -1047,8 +1047,8 @@ def test_main(self, mocker):
         diagnostics_mock.execute.return_value = (None, None, None)
 
         mocker.patch(MODULE_PATH + 'get_argument_spec', return_value={})
-        mocker.patch(MODULE_PATH + 'idrac_auth_params', {})
-        mocker.patch(MODULE_PATH + 'AnsibleModule', return_value=module_mock)
+        # mocker.patch(MODULE_PATH + 'idrac_auth_params', {})
```
Comment (rajshekarp87) [inline tests/unit/plugins/modules/test_idrac_diagnostics.py]:
```suggestion

```

**Comment by ABHISHEK-SINHA10** [comment]
@jagadeeshnv  Update the version and copyright year in test files too, there'll be mismatch in module and test file.

File: plugins/modules/ome_template_info.py [hunk 11]
```diff
@@ -105,8 +105,8 @@
 
 import json
 from ssl import SSLError
-from ansible.module_utils.basic import AnsibleModule
-from ansible_collections.dellemc.openmanage.plugins.module_utils.ome import RestOME, ome_auth_params
+
```
Comment (ABHISHEK-SINHA10) [inline plugins/modules/ome_template_info.py]: New line is being added to all the file ome modules before OmeAnsibleModule import, just check once.

**Review by rajshekarp87 — DISMISSED** [review]

**Review by rajshekarp87 — APPROVED** [review]

**Review by sachin-apa — APPROVED** [review]

**Pull request merged**